### ISSUE TITLE
Question about normalization of coeff_w and coeff_x.

### ISSUE THREAD

**Issue by veason** [wy1iu/sphereface#62]
Hi, @wy1iu 
In backward step, coeff_x and coeff_w are both normalized (coeff_x ^2 + coeff_w ^ 2 = 1). This implement is different from paper "large-margin softmax loss for convolutional neural networks". What is the use of the normalization? Is it a trick?

**Comment by veason** [comment]
I saw the backward gradient normalization in **note**. It's pretty impressive. THX.
